docs/learn.md

```diff
@@ -1,3 +1,153 @@
+
+Melange helps developers and companies bring the advantages of the OCaml
+platform to users of the Web platform in a way that makes it easy for developers
+to integrate with both ecosystems.
+
+### A bit of history
+
+To better understand where Melange comes from, it might help to go through some
+of the related projects that have appeared over the last decade or so.
+
+#### Js_of_ocaml
+
+[Js_of_ocaml](https://github.com/ocsigen/js_of_ocaml/) is another OCaml to
+JavaScript compiler that was made public in 2011. In [the presentation
+paper](https://www.irif.fr/~balat/publications/vouillon_balat-js_of_ocaml.pdf)
+published in 2013, it's explicitly mentioned that one of its design goals was to
+remain as compatible as possible with the OCaml compiler, without requiring a
+lot of maintenance work, as the OCaml community was not excessively large at the
+time.
+
+To achieve this goal, Js_of_ocaml picks the bytecode generated by OCaml [batch
+compilation](https://ocaml.org/manual/comp.html) and generates JavaScript from
+it. OCaml bytecode has a very stable interface, so Js_of_ocaml can easily
+upgrade to new versions of the compiler. Due to this design decision, it can
+also remain compatible with most of the OCaml ecosystem, as long as the tools or
+libraries don't rely on C code.
+
+The downside of using bytecode is that it gets harder to communicate with
+existing JavaScript code. For example, a Js_of_ocaml string is not compiled as a
+JavaScript string, so some conversion is needed for this and other types.
+
+Another downside is that the resulting JavaScript is hard to read, as it is
+converted from a low-level representation like bytecode.
+
+#### BuckleScript
+
+Then, in 2016, Bob Zhang suggests [on a Js_of_ocaml repository
+issue](https://github.com/ocsigen/js_of_ocaml/issues/338) the possibility to
+start converting to JavaScript from an earlier stage of the compilation process,
+instead of using bytecode. This proposal fundamentally diverges from Js_of_ocaml
+original design and goals, so he starts working on what will become
+BuckleScript.
+
+BuckleScript gets some inspiration from Js_of_ocaml, for example in the way that
+JavaScript objects are represented with `Js.t`. But it differs from Js_of_ocaml
+in many ways: it can generate more readable and lighter code. It also generates
+one `.js` file per module, which makes it easier to integrate with existing
+JavaScript codebases. BuckleScript puts a big emphasis on communicating with
+JavaScript code through a rich collection of attributes applied to `external`
+primitives.
+
+#### Reason
+
+Around the same year, a project called [Reason](https://reasonml.github.io/)
+appears at Facebook. Led by Jordan Walke, the idea is to create an alternate
+syntax for OCaml that is closer to C and JavaScript. Even if Reason has no take
+on which platform the code is deployed —native applications binaries, or web
+applications using JavaScript as a target language—, BuckleScript adds first
```

Comment:
this needs a space after the emdash.

Reply:
You mean space on both sides around em dashes? I thought that was [optional](https://www.merriam-webster.com/words-at-play/em-dash-en-dash-how-to-use):

> Spacing around an em dash varies. Most newspapers insert a space before and after the dash, and many popular magazines do the same, but most books and journals omit spacing, closing whatever comes before and after the em dash right up next to it. This website prefers the latter, its style requiring the closely held em dash in running text.

Reply:
Yeah I've seen you use this style in other places too. No strong opinions, then.